cmake/modules/AddPureSwift.cmake

```diff
@@ -16,12 +15,18 @@ endfunction()
 function(force_target_link_libraries TARGET)
   target_link_libraries(${TARGET} ${ARGN})
 
-  cmake_parse_arguments(ARGS "PUBLIC;PRIVATE;INTERFACE" "" "" ${ARGN})
-  force_add_dependencies(${TARGET} ${ARGS_UNPARSED_ARGUMENTS})
+  if(NOT CMP0157_IS_NEW)
```

Comment:
Didn't this get fixed in 3.26 or something?
 We can use `CMAKE_VERSION` with one of `VERSION_LESS`, `VERSION_GREATER`, `VERSION_EQUAL`, `VERSION_LESS_EQUAL`, or `VERSION_GREATER_EQUAL` to do the comparison directly on the version itself rather than using the presence of the CMP0157 policy.

Reply:
I'm not really sure, tbh I didn't tried just removing this without enabling `CMP0157`.

Is just checking `CMAKE_VERSION` enough? As far as I tried, `CMP0157` had to be set `NEW` to remove these workarounds. Though I didn't tried removing each workaround.

Reply:
I fixed the missing dependency edge on the swiftmodule here: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/8084.
It should be fixed in anything newer than CMake 3.26. Versions older that 3.29 may have unnecessary rebuilds though, but that shouldn't be different than the forced rebuild anyway.